backend/tests/buildings.test.js

```diff
@@ -10,4 +10,39 @@ describe('Buildings', () => {
         .expect(200, done);
     });
   });
+
+  describe('GET /kingdom/buildings/', () => {
+    it('should return 200 when getting all buildings (with slash)', (done) => {
+      request(app)
+        .get('/kingdom/buildings/')
+        .expect('Content-Type', /json/)
+        .expect(200, done);
+    });
+  });
+
+  describe('GET /kingdom/buildings/non-number', () => {
+    it('should return 400 when the building id is not a number', (done) => {
+      request(app)
+        .get('/kingdom/buildings/non-number')
+        .expect(400, done);
+    });
+  });
+
+  describe('GET /kingdom/buildings/0', () => {
```

Comment:
I think you should handle this in the router, because no building will have an id of 0, and it should return with 404. Handle this case in the router and test it too.